### ISSUE TITLE
[vtk&&vtk-m] build with cuda12 failure

### ISSUE THREAD

**Issue by Bzz2745** [microsoft/vcpkg#29841]
**Host Environment**

- OS: Windows 11
- Compiler: MSVC vc143

**To Reproduce**

`./vcpkg install vtk-m[core,cuda,double,mpi,tbb]:x64-windows --clean-after-build`

**Failure logs**

Building vtk-m[core,cuda,double,mpi,tbb]:x64-windows...
-- Feature CUDA forces static build!
-- Using cached vtk-vtk-m-902fdac6fafb6358ce88f8747d55e2c0715241f1-1.tar.gz.
-- Extracting source C:/vcpkg/downloads/vtk-vtk-m-902fdac6fafb6358ce88f8747d55e2c0715241f1-1.tar.gz
-- Using source at C:/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j13
    Working Directory: C:/vcpkg/buildtrees/vtk-m/x64-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\vtk-m\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/vtk-m/portfile.cmake:50 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)

error: building vtk-m:x64-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+vtk-m
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[vtk-m]+Build+error
Include '[vtk-m] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 54cc53c43 2023-02-23 (11 hours ago)

**Additional context**

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10823959/install-x64-windows-dbg-out.log)

one more thing, I found the same error in this link : https://bugs.gentoo.org/894646

**Comment by Bzz2745** [comment]
![image](https://user-images.githubusercontent.com/68807654/221177895-5a83b0ee-10df-4797-b70e-402b317d626e.png)
I will try this.

**Comment by Bzz2745** [comment]
> ![image](https://user-images.githubusercontent.com/68807654/221177895-5a83b0ee-10df-4797-b70e-402b317d626e.png) I will try this.

It dosen't work,here is the log:
 
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10824709/install-x64-windows-dbg-out.log)

**Comment by MonicaLiu0311** [comment]
Win11
VS 2022
```
PS E:\vcpkg02> ./vcpkg version
vcpkg package management program version 2023-02-16-12e657924d99511514c0287ca5ce46882d3657c7

See LICENSE.txt for license information.
```
Thanks for your feedback, but I can't reproduce your issue.

**Comment by netcorefan1** [comment]
I just want to confirm what the user stated in this issue. I'm also on Cuda 12 and when I try to install the feature (`vcpkg install vtk-m[cuda]:x64-windows-release --recurse`), I get the same error.
VS 2022 17.5.1
MSVC v143 (marked as latest from the VS installer)

**Comment by Bzz2745** [comment]
> 

My msvc version is the same as yours.

**Comment by Bzz2745** [comment]
> 我只想确认用户在此问题上的陈述。 I'm also on Cuda 12 and when I try to install the feature (`vcpkg install vtk-m[cuda]:x64-windows-release --recurse`), I get the same error. VS 2022 17.5.1 MSVC v143 (marked as latest from the VS installer)

My msvc version is the same as yours.

**Comment by netcorefan1** [comment]
> > 我只想确认用户在此问题上的陈述。 I'm also on Cuda 12 and when I try to install the feature (`vcpkg install vtk-m[cuda]:x64-windows-release --recurse`), I get the same error. VS 2022 17.5.1 MSVC v143 (marked as latest from the VS installer)
> 
> My msvc version is the same as yours.

I have just run `git pull` and downloaded all updates, but I noticed that `vtk-m[core,cuda]` is still version `1.9.0#1` resulting in the same error and I'm wondering if @MonicaLiu0311 is having troubles to apply the fix.
I'm using vtk as part of OpenVc. Is it the same for you?
I'm trying to compile OpenCv from several days without success due to the issue on this topic and other issues I found in other related packages (like for example [OpenBlas](https://github.com/microsoft/vcpkg/issues/29994)). One can simply skip the non working packages and recompile once the fixes are released. Problem is that recompiling everything from scratch each time a package is fixed is extremely slow!
